### ISSUE TITLE
[Remove Free Workspace Creation] Keep old NewDot clients from Creating Free Workspaces

### ISSUE THREAD

**Issue by tgolen** [Expensify/App#37154]
- Held on https://github.com/Expensify/App/issues/37152
- Held on https://github.com/Expensify/App/issues/37153

Coming from the design doc section [here](https://docs.google.com/document/d/16ptT3EbBMHCP7TzUBlRADWuBASo3pL12d_Z0wYr38c4/edit#bookmark=id.xtjrg38hce9w)

It could be that old NewDot clients would still have old code allowing them to create free workspaces. To prevent that, there will be some logic added to the `CreateWorkspace` API command.

- Update `api.php` so that if the client passes a policy type of "free", then set the type to "team" [here](https://github.com/Expensify/Web-Expensify/blob/1c355c41dc0f10a47e98e30ab8784d14b1942441/api.php#L761-L779)
- Add a Log::info() when that happens so we can detect when old clients are making bad requests

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01dfd3c0fbfc17add5</li>
        <li>Upwork Job ID: 1761071438657032192</li>
        <li>Last Price Increase: 2024-02-27</li>
    </ul>
</details>

**Comment by tgolen** [comment]
I am unassigning @mollfpr since they cannot review the internal PR on the API side.

**Comment by tgolen** [comment]
# Daily Update
- This issue is on HOLD
- The [Web-E PR](https://github.com/Expensify/Web-Expensify/pull/41030) is created which is also on HOLD

# Next Steps
- Wait for linked issues to be completed
- Take issue off HOLD and get PR merged

#### ETA
- Taken off HOLD by Friday Mar 8
- Merged and deployed by Wednesday Mar 13

**Comment by tgolen** [comment]
# Daily Update
- This issue is still on HOLD
- The two dependent issues have been taken off HOLD today, so they are in motion

### Next Steps

- Wait for linked issues to be completed
- Take issue off HOLD and get PR merged

#### ETA
Taken off HOLD by Monday Mar 11
Merged and deployed by Wednesday Mar 13

**Comment by tgolen** [comment]
# Daily Update
- This issue is still on HOLD
- https://github.com/Expensify/App/issues/37152 has an accepted proposal but doesn't have a PR yet
- https://github.com/Expensify/App/issues/37153 is still evaluating proposals

### Next Steps
No change

#### ETA
No change

**Comment by tgolen** [comment]
# Daily Update
- This issue is still on HOLD
- https://github.com/Expensify/App/pull/38023 needs to be reviewed/merged/deployed
- https://github.com/Expensify/App/pull/38191 needs to be reviewed/merged/deployed

### Next Steps
- Wait for those two PRs to be deployed

### ETA
- Maybe ready to be taken off HOLD by March 20
- Have this issued finished and deployed by March 22
- I'll be OOO that week, so there could be some delay in getting it out, but [the Web-E PR](https://github.com/Expensify/Web-Expensify/pull/41030) has already been reviewed and approved

**Comment by tgolen** [comment]
# Daily Update
- This issue is still on HOLD
- https://github.com/Expensify/App/pull/38023 is actively being reviewed
- https://github.com/Expensify/App/pull/38191 is actively being reviewed

### Next Steps
- Waiting for those two PRs to be deployed to production

#### ETA
- No change

**Comment by tgolen** [comment]
# Daily Update
- This issue is still on HOLD
- https://github.com/Expensify/App/pull/38023 doesn't look like it has any changes since yesterday
- https://github.com/Expensify/App/pull/38191 doesn't look like it has any changes since yesterday

### Next Steps
- Wait for those to PRs to be deployed to production
- @danielrvidal will be watching this while I am OOO next week to merge [the Web-E PR](https://github.com/Expensify/Web-Expensify/pull/41030) once the two PRs above have been deployed to production

#### ETA (REVISED)
- Ready to be taken off HOLD by March 22
- Have this issue finished and deployed by March 26

**Comment by danielrvidal** [comment]
Sounds good, I'll check in on this on Monday!

**Comment by danielrvidal** [comment]
# Daily Update
- This issue is still on HOLD
- https://github.com/Expensify/App/pull/38023 doesn't look like it has any changes since yesterday
- https://github.com/Expensify/App/pull/38191 doesn't look like it has any changes since yesterday

### Next Steps
- Wait for those to PRs to be deployed to production
  - I'm working with contributors on the first. 
  - @MariaHCD is reviewing this tomorrow to unblock the second: https://github.com/Expensify/Web-Expensify/pull/41317
- I will be watching this while I am OOO next week to merge [the Web-E PR](https://github.com/Expensify/Web-Expensify/pull/41030) once the two PRs above have been deployed to production

#### ETA (REVISED)
- Ready to be taken off HOLD by March 22
- Have this issue finished and deployed by March 26

**Comment by danielrvidal** [comment]
We ended up hitting a speed bump because we think we accidentally turned off the emails for NewDot, but I think we have a good path forward so will be creating next steps today.

**Comment by tgolen** [comment]
# Daily Update
- #38191 has been merged 🎉 
- That only leaves one remaining PR until the hold can be removed: #38023

### Next Steps
- Wait for final [App PR](https://github.com/Expensify/App/pull/38023) to be merged
- Take the [Web-E PR](https://github.com/Expensify/Web-Expensify/pull/41030) off HOLD and merge/deploy it

#### ETA
- Ready to be taken off HOLD by March 29
- Have this issue finished and deployed by April 3

**Comment by tgolen** [comment]
# Daily Update
- Still on hold for #38023 

### Next Steps
- Same

#### ETA
- Same

**Comment by tgolen** [comment]
# Daily Update
- Same

### Next Step
- Same

#### ETA
- Same

**Comment by tgolen** [comment]
- Same

**Comment by tgolen** [comment]
# Daily Update
- This has been taken off HOLD because it really doesn't matter if the welcome note is up to date or not
- The [Web-E PR](https://github.com/Expensify/Web-Expensify/pull/41030) has been merged

### Next Steps
- Wait for it to be deployed and then this issue will close out

#### ETA
- It will probably deploy to production on Monday, April 8

**Comment by danielrvidal** [comment]
Woo! Thanks @tgolen
